ConnectorTeklaStructures/ConnectorTeklaStructuresShared/StreamStateManager/StreamStateManager.cs

```diff
@@ -1,142 +1,210 @@
+using System;
 using DesktopUI2.Models;
 using Speckle.Newtonsoft.Json;
 using System.Collections.Generic;
 using System.IO;
 using System.Text;
+using Speckle.Core.Logging;
 using Tekla.Structures.Model;
 
 namespace ConnectorTeklaStructures.Storage;
 
 public static class StreamStateManager
 {
-  private static string _speckleFilePath;
+  private static string s_speckleFilePath;
 
   public static List<StreamState> ReadState(Model model)
   {
     var strings = ReadSpeckleFile(model);
-    if (strings == "")
+    if (string.IsNullOrEmpty(strings))
     {
       return new List<StreamState>();
     }
     try
     {
       return JsonConvert.DeserializeObject<List<StreamState>>(strings);
     }
-    catch
+    catch (JsonException ex)
     {
+      SpeckleLog.Logger.Warning(
+        "Failed to deserialize saved stream state list: " + $"{ex.Message}. Resetting to an empty list."
+      );
       return new List<StreamState>();
     }
   }
 
   /// <summary>
-  /// Writes the stream states to the <TeklaStructuresModelName>.txt file in speckle folder
+  /// Writes the stream states to the {TeklaStructuresModelName}.txt file in speckle folder
   /// that exists or is created in the folder where the TeklaStructures model exists.
   /// </summary>
   /// <param name="model"></param>
   /// <param name="streamStates"></param>
   public static void WriteStreamStateList(Model model, List<StreamState> streamStates)
   {
-    if (_speckleFilePath == null)
+    if (s_speckleFilePath == null)
     {
       GetOrCreateSpeckleFilePath(model);
     }
 
-    FileStream fileStream = new(_speckleFilePath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
+    if (s_speckleFilePath == null)
+    {
+      return;
+    }
+
+    FileStream fileStream = new(s_speckleFilePath, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
+
+    if (!fileStream.CanWrite)
+    {
+      return;
+    }
+
     try
     {
-      using (var streamWriter = new StreamWriter(fileStream))
+      using var streamWriter = new StreamWriter(fileStream);
+      string streamStateString = JsonConvert.SerializeObject(streamStates);
+
+      if (string.IsNullOrEmpty(streamStateString))
       {
-        streamWriter.Write(JsonConvert.SerializeObject(streamStates) as string);
-        streamWriter.Flush();
+        return;
       }
+
+      streamWriter.Write(streamStateString);
+      streamWriter.Flush();
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      SpeckleLog.Logger.Error("Failed to write stream state list to file");
```

Comment:
Does the caller get any feedback?

Reply:
Actually, DUI catches errors when calling this, so can rethrow instead.